schema_test.go

```diff
@@ -380,3 +381,119 @@ func TestIncorrectRef(t *testing.T) {
 	assert.Nil(t, s)
 	assert.Equal(t, "Object has no key 'fail'", err.Error())
 }
+
```

Comment:
The tests regarding `format` should be in `format_checkers_test.go`, `schema_test.go` has become a place where all miscellaneous tests end up, but these tests don't have to be in there.